### PR TITLE
[Serve] Remove old grpc code

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

1. Remove is_driver_deployment API
2. Remove the grpc driver
3. Remove driver deployment.
4. Rmove disable_httpproxy flag
5. Remove related unit tests
6. Remove old gRPC proto schema

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/drivers.py [hunk 1]
```diff
@@ -118,104 +111,3 @@ async def get_intermediate_object_refs(self) -> Dict[str, Any]:
...
-        """Start gRPC Server"""
-        logger.info(
-            "Starting gRPC server with on node:{} "
-            "listening on port {}".format(ray.util.get_node_ip_address(), self.port)
-        )
-        address = "[::]:{}".format(self.port)
-        try:
-            # Depending on whether RAY_USE_TLS is on, `add_port_to_grpc_server`
-            # can create a secure or insecure channel
-            self.grpc_port = add_port_to_grpc_server(self.server, address)
```
Comment (sihanwang41) [inline python/ray/serve/drivers.py:155]: @GeneDer  do we support TLS (secure channel) in the current grpc proxy?

Comment (GeneDer) [reply python/ray/serve/drivers.py:155]: Right now it's just using unsecured port https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/proxy.py#L1617 But adding support should be pretty easy.

**Comment by edoakes** [comment]
@sihanwang41 could you please separate (7) out into its own PR so this is purely removing the existing stuff?

**Review by edoakes — APPROVED** [review]

**Comment by sihanwang41** [comment]
> @sihanwang41 could you please separate (7) out into its own PR so this is purely removing the existing stuff?

done

**Review by zcin — APPROVED** [review]

**Pull request merged**